GAE/src/org/waterforpeople/mapping/dataexport/RawDataSpreadsheetImporter.java

```diff
@@ -434,18 +445,18 @@ private Integer parseRepeatsForInstance(InstanceData instanceData,
         // 8. SurveyalTime - ignored duplicate
         // 9 - N. Questions
 
-        int instanceIdColumnIndex = metadataColumnHeaderIndex2.get(SURVEY_INSTANCE_COLUMN_KEY);
-        int repeatIterationColumnIndex = metadataColumnHeaderIndex2.get(REPEAT_COLUMN_KEY);
+        int identifierColumnIndex = metadataColumnHeaderIdx.get(DATAPOINT_IDENTIFIER_COLUMN_KEY);
+        int repeatIterationColumnIndex = metadataColumnHeaderIdx.get(REPEAT_COLUMN_KEY);
 
-        String SurveyInstanceId = instanceData.surveyInstanceDto.getKeyId().toString();
+        String dataPointIdentifier = instanceData.surveyInstanceDto.getSurveyedLocaleIdentifier();
         //Find first row matching this, or give up
         int rowIx = currentRowIndex;
         Row row;
         while (true) {
             row = repSheet.getRow(rowIx);
             if (row != null
-                    && row.getCell(instanceIdColumnIndex) != null
-                    && row.getCell(instanceIdColumnIndex).getStringCellValue().equals(SurveyInstanceId)) { //found!
+                    && row.getCell(identifierColumnIndex) != null
+                    && row.getCell(identifierColumnIndex).getStringCellValue().equals(dataPointIdentifier)) { //found!
                 break;
             } else {
                 if (isEmptyRow(row)) { // a row without any cells defined
```

Comment:
My understanding is that we are scanning the datapoint identifier column to find matching rows, but if we use this we end up scanning the whole row perhaps needlessly.  Would it be enough to search here for `isEmptyCell(identifierColumnIndexCell)`? And just ignore those that don't have an identifier?

Reply:
I thought this was slightly more efficient, but sure.